cinema/views.py

```diff
@@ -34,6 +39,32 @@ class MovieViewSet(viewsets.ModelViewSet):
     queryset = Movie.objects.all()
     serializer_class = MovieSerializer
 
+    @staticmethod
+    def _params_to_ints(qs):
+        """Converts a list of string IDs to a list of integers"""
+        return [int(str_id) for str_id in qs.split(",")]
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        actors = self.request.query_params.get("actors")
+
+        if actors:
+
+            actors_ids = self._params_to_ints(actors)
+            queryset = Movie.objects.filter(actors__id__in=actors_ids)
+
+        genres = self.request.query_params.get("genres")
+        if genres:
+            genres_ids = self._params_to_ints(genres)
+            queryset = Movie.objects.filter(genres__id__in=genres_ids)
+
+        title = self.request.query_params.get("title")
+        if title:
+            title_str = title
+            queryset = Movie.objects.filter(title__icontains=title_str)
+        return queryset.distinct()
```

Comment:
Use a queryset to filter objects rather than Movie.objects. Because every time you filter by Movie.objects, your previous filter will not work. For example, if you filter by actors and genres, then you will filter by actors first, and store that result in a query set, and then filter by genre, not considering the filter by actors, because now you are only filtering by genre
```suggestion
        if actors:

            actors_ids = self._params_to_ints(actors)
            queryset = queryset.filter(actors__id__in=actors_ids)

        genres = self.request.query_params.get("genres")
        if genres:
            genres_ids = self._params_to_ints(genres)
            queryset = queryset.filter(genres__id__in=genres_ids)

        title = self.request.query_params.get("title")
        if title:
            title_str = title
            queryset = queryset.filter(title__icontains=title_str)
        return queryset.distinct()
```

Reply:
Thank you for the useful information.